libuserhelper.cpp

```diff
@@ -110,6 +110,37 @@ bool LibUserHelper::addUserToGroup(const QString &user, const QString &group)
     return rv;
 }
 
+bool LibUserHelper::removeUserFromGroup(const QString &user, const QString &group)
+{
+    struct lu_error *error = nullptr;
+    struct lu_context *context = lu_start(NULL, lu_user, NULL, NULL, NULL, NULL, &error);
+    if (!context) {
+        qCWarning(lcSUM) << "Error creating context";
+        return false;
+    }
+
+    bool rv = true;
+    struct lu_ent *ent = lu_ent_new();
+    if (lu_group_lookup_name(context, group.toLocal8Bit(), ent, &error)) {
+        GValue *value = g_slice_new0(GValue);
+        g_value_init(value, G_TYPE_STRING);
+        g_value_set_string(value, user.toUtf8());
```

Comment:
Are you sure this is safe?

`QString::toUtf()` returns `QByteArray` (which contains UTF-8 data converted from iternal UTF-16 representation). Then `QByteArray` gets cast to `const char*` which returns pointer to the internal buffer. That's the value which will be passed to `g_value_set_string()`. Now that compiler has the value it can destroy that temporary `QByteArray` which is no longer needed (along with the data which the pointer is pointing to). Are you sure that `QByteArray` is going to be destroyed **after** `g_value_set_string()` call and not before?

I'm quite sure I've seen similar cases when such a temporary `QByteArray` was being destroyed too early and the function received pointer to already deallocated memory. I would keep `QByteArray` on stack to prevent its buffer from being freed too early.

The same applies to `QString::toLocal8Bit()` - that one also returns `QByteArray`.

Reply:
Specs say temporary object is valid through the full-expression, that should contain the whole line. Haven't seen any problems with this yet but lets keep in mind.

Reply:
I prefer to keep `QByteArray` on stack in such cases.

Reply:
@monich me too, but mostly because otherwise I have to think about how did it go again... The real problems I've seen/fixed have been of kind where the pointer is stored for later use, like if it were like:

    const char *text = user.toUtf8();
    g_value_set_string(value, text);